### DISCUSSION TITLE
Hypothetical pure C++ implementation of the Zenoh protocol

### DISCUSSION THREAD

**Discussion by doganulus** [eclipse-zenoh/roadmap#158] [Q&A] [answered]
Given the API reference for the version 1.0:
https://zenoh-cpp.readthedocs.io/en/1.0.0.7/

Hypothetically, can the same public API be used to implement the Zenoh protocol in pure C++ without the Rust backend? Does the API depend on the Rust implementation in any tighter and deeper manner? I think it must be possible but some rough edges might be there.

Actually, I wonder if a standard high-level comm library (like a C++ standard library) can be defined starting from this point.

**Comment by Mallets** [comment, marked as answer]
Hello, the API can use zenoh-pico as a backend. No need to necessarily rely on the Rust backend (zenoh-c). See https://github.com/eclipse-zenoh/zenoh-cpp

**Reply by doganulus** [reply]
Thank you @Mallets. Apparently, I missed the fact that zenoh-pico is a native C implementation. I will look at it more closely. It is actually great that you have two compatible implementations.
